### DISCUSSION TITLE
My experience building via cargo on WSL Ubuntu

### DISCUSSION THREAD

**Discussion by iwyatt** [tramhao/termusic#142] [General]
In Ubuntu WSL  using  `cargo install termusic termusic-server`:

TLDR: Needed to manually install 'libdbus-sys', `sudo apt install libasound2-dev`,  `sudo apt install protobuf-compiler` before I received a successful compile message. And after all that, when I ran `termusic-server` and it failed, I realized that Ubuntu WSL doesn't even have a sound device to playback music on. /facepalm . 

I might try cross compiling it to Windows 10 instead before loading up a full Ubuntu desktop to try it.

```
pkg_config failed: `PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" PKG_CONFIG_ALLOW_SYSTEM_LIBS="1" "pkg-config" "--libs" "--cflags" "dbus-1" "dbus-1 >= 1.6"` did not exit successfully: exit status: 1
  error: could not find system library 'dbus-1' required by the 'libdbus-sys' crate
```

per [related github issue](https://github.com/greshake/i3status-rust/issues/194) tried `sudo apt install libdbus-1-dev`

and then tried cargo install again but the build failed with the following error:
```
thread 'main' panicked at '`PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" PKG_CONFIG_ALLOW_SYSTEM_LIBS="1" "pkg-config" "--libs" "--cflags" "alsa"` did not exit successfully: exit status: 1
  error: could not find system library 'alsa' required by the 'alsa-sys' crate
```
per a [hint in this issue comment](https://github.com/Spotifyd/spotifyd/issues/659#issuecomment-675336593) I ran `sudo apt install libasound2-dev` and attempted to compile again:

During this build a warning and error were thrown about `PROTOC` not being available resulting in the error `error: failed to run custom build command for `termusic-playback v0.7.11`` being thrown. It instructed me to `apt-get install protobuf-compiler` but since the compiler continued to chug through I didn't have a chance to actually do anything about it until finally failed completely.

So I did `sudo apt install protobuf-compiler` and again, for the third time, ran `cargo install termusic termusic-server`.

On this attempted I finally received the message `Summary Successfully installed termusic, termusic-server!`

When I ran `termusic-server`, it failed with `ALSA lib confmisc.c:855:(parse_card) cannot find card '0'`.

Duh. There isn't a sound device for it to initialize.

**Comment by tramhao** [comment]
Currently the build script for ci is working for windows. I build it under msys2 mingw64 environment and works. Only problem now is album photo . It works but not very good. If turn off cover, everything is fine.

**Comment by iwyatt** [comment]
Thanks!
